src/framework/components/anim/controller.js

```diff
@@ -238,8 +238,8 @@ Object.assign(pc, function () {
             get: function () {
                 var playable = true;
                 var i;
-                for (i = 0; i < this._states.length; i++) {
-                    if (!this._states[i].playable) {
+                for (i = 0; i < this.states.length; i++) {
+                    if (!this._states[this.states[i]].playable) {
```

Comment:
This does look a bit odd. Are `this.states` really ids or names or something?

Reply:
Yeah the controller exposes a list of available state names to the component but they're stored internally as AnimState objects. I could loop over `this._states` keys instead here if it makes things clearer.

Reply:
Using `this._stateNames` here instead.